contracts/lazyclaim/ERC721LazyClaim.sol

```diff
@@ -0,0 +1,159 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.0;
+
+/// @author: manifold.xyz
+
+import "@manifoldxyz/creator-core-solidity/contracts/core/IERC721CreatorCore.sol";
+import "@manifoldxyz/libraries-solidity/contracts/access/AdminControl.sol";
+import "@manifoldxyz/creator-core-solidity/contracts/extensions/ICreatorExtensionTokenURI.sol";
+
+import "@openzeppelin/contracts/security/ReentrancyGuard.sol";
+import "@openzeppelin/contracts/utils/cryptography/MerkleProof.sol";
+import "@openzeppelin/contracts/interfaces/IERC165.sol";
+
+import "./IERC721LazyClaim.sol";
+
+/**
+ * Lazy claim with optional whitelist ERC721 tokens
+ */
+contract ERC721LazyClaim is IERC165, IERC721LazyClaim, ICreatorExtensionTokenURI, ReentrancyGuard {
+  mapping(address => uint) public claimCounts;
+  mapping(address => mapping(uint => Claim)) public claims;
+  mapping(address => mapping(uint => uint)) public mintsPerClaim;
+  mapping(address => mapping(uint => mapping(address => uint))) public mintsPerWallet;
+  mapping(address => mapping(uint => uint)) public tokenClaims;
+
+  function supportsInterface(bytes4 interfaceId) public view virtual override(IERC165) returns (bool) {
+    return interfaceId == type(IERC721LazyClaim).interfaceId ||
+      interfaceId == type(ICreatorExtensionTokenURI).interfaceId ||
+      interfaceId == type(IERC165).interfaceId;
+  }
+
+  modifier creatorAdminRequired(address creatorContractAddress) {
+    // This is shared, not single-creator. So we must ensure that the initializer is an admin of the contract
+    AdminControl creatorCoreContract = AdminControl(creatorContractAddress);
+    require(creatorCoreContract.isAdmin(msg.sender), "Wallet is not an administrator for contract");
+    _;
+  }
+
+  // Initialize
+  function initializeClaim(
+    address creatorContractAddress,
+    bytes32 merkleRoot,
+    string calldata uri,
+    uint totalMax,
+    uint walletMax,
+    uint startDate,
+    uint endDate
+  ) external creatorAdminRequired(creatorContractAddress) {
+    // Get the index for the new claim
+    uint newIndex = claimCounts[creatorContractAddress];
+    claimCounts[creatorContractAddress] = newIndex + 1;
+
+    // Create the claim
```

Comment:
Do input sanity checking first! (require statements)

Reply:
Done, though only 1 check on initialize (endDate > startDate).

More checks for overwrite, which are also implemented.